### PR TITLE
Rewards architectural improvements.

### PR DESCRIPTION
**Pull request by szilardszaloki**
Resolves https://github.com/brave/brave-browser/issues/29557.

# What does this PR do?
- removes the proxy/bridge layers (and any related abstractions) on both sides of the Mojom message pipe between the browser process and the utility process that were only there to convert between legacy callback types (`std::function<>`s) and `base::OnceCallback<>`s
- changes responsibilities of the affected components:
   - `LedgerFactory` (previously `BatLedgerService`): `BatLedgerService` was given too much responsibility, as besides creating the ledger, it was able to set/get the ledger's global variables — `LedgerFactory` can only create the ledger.
   - `Ledger` (previously `BatLedger`): same functionality as before, plus maintaining its own global variables (which the `BatLedgerService` used to do). Also renaming, as the name `Ledger` became available.
   - `LedgerClient` (previously `BatLedgerClient`): no change in functionality, just renaming (as the name `LedgerClient` freed up).
- `ledger_impl.h` now only forward declares its subcomponents. Previously you could not include `ledger_impl.h` in most places, e.g. putting it in `uphold.h` would result in the following circular dependency:
   ```
   In file included from ../../brave/components/brave_rewards/core/uphold/uphold_card.cc:6:
   In file included from ../..\brave/components/brave_rewards/core/uphold/uphold_card.h:13:
   In file included from ../..\brave/components/brave_rewards/core/uphold/uphold.h:21:
   ../..\brave/components/brave_rewards/core/ledger_impl.h(307,19): error: no type named 'Uphold' in namespace 'ledger::uphold'
     ledger::uphold::Uphold* uphold() const;
     ~~~~~~~~~~~~~~~~^
   ../..\brave/components/brave_rewards/core/ledger_impl.h(485,35): error: no member named 'Uphold' in namespace 'ledger::uphold'
     std::unique_ptr<ledger::uphold::Uphold> uphold_;
                     ~~~~~~~~~~~~~~~~^
   ```
   That's because `ledger_impl.h` itself includes `uphold.h`. Above is true for all of `LedgerImpl`'s subcomponents (of which we currently have 13). These circular dependencies can be broken by only forward declaring the subcomponents in `ledger_impl.h`, and making all subcomponents include only those other subcomponents that they actually use, that is, every subcomponent now "pays for what they use" — e.g. they don't have to include `state.h` if they don't need access to states, etc.
- introduces `ledger_callbacks.h`, which is a common place for callbacks used by the ledger
- removes any references from the browser process that are internal to the ledger (e.g. the ledger's global variables, callback types, etc. should not be used in the browser process)
- removes `support_lazy_serialization = true` from `components/services/bat_ledger/public/interfaces`, as this flag disables Mojom message validation if the messages go between endpoints in the same process ([spec](https://source.chromium.org/chromium/chromium/src/+/main:mojo/public/tools/bindings/mojom.gni;l=278-283))
- cleans up the massive overuse of `std::unique_ptr<>`s in unit tests
- uses `testing::MockFunction<>`s (for `std::function<>`s) and `base::MockCallback<>`s (for `base::OnceCallback<>`s) to ensure that no asynchronous expectations are missed

A remarkable side-effect of the above changes is that iOS now becomes a fully functional Mojom client of Rewards. Also, no iOS-only APIs on the `Ledger` interface anymore, as now it's Mojom-generated (see below).

## APIs present on `ledger::Ledger`, but not on `mojom::BatLedger`:
- unused:
   - `GetAllBalanceReports()` ❌
- used by some leftover code on iOS (`brave_ledger.h`/`.mm`):
   - `GetDrainStatus()` ❌ (used by `drainStatusForDrainId`)
   - `ProcessSKU()` ❌(used by `processSKUItems`)

Drain status changes:
- we no longer need `ledger::endpoint::promotion::GetDrain` (the implementation of `GET /v1/promotions/drain/drain-id`):
   - `components/brave_rewards/core/endpoint/promotion/get_drain.h` ❌
   - `components/brave_rewards/core/endpoint/promotion/get_drain.cc` ❌

SKU changes:
- we don't use `ledger::sku::SKUMerchant` anymore:
   - `components/brave_rewards/core/sku/sku_merchant.h` ❌
   - `components/brave_rewards/core/sku/sku_merchant.cc` ❌
- we no longer need a common `ledger::sku::SKU` abstraction, as there's only one implementation now — `ledger::sku::SKUBrave`:
   - `components/brave_rewards/core/sku/sku.h` ❌ 
- hence we don't need a factory for SKU implementations either:
   - `components/brave_rewards/core/sku/sku_factory.h` ❌ 
   - `components/brave_rewards/core/sku/sku_factory.cc` ❌ 
- but then we can also rename `ledger::sku::SKUBrave` to `ledger::sku::SKU`:
   - `components/brave_rewards/core/sku/sku_brave.h` `==>` `components/brave_rewards/core/sku/sku.h`
   - `components/brave_rewards/core/sku/sku_brave.cc` `==>` `components/brave_rewards/core/sku/sku.cc`

## Header inclusion best practices:
- include `ledger_callbacks.h`, if you only need the callback types (typically in headers)
- include `ledger_impl.h`, if besides the callback types you also need the ledger (or its global variables), but not its subcomponents (typically in implementation files)
- if your implementation requires a certain subcomponent, include `ledger_impl.h` + the subcomponent itself (e.g. if you need `state::State`, make sure to also include `state.h`)

# Why are we making these changes?
For the sake of comparison, let's assume that we need to add a function to the Rewards service (browser process), which then can be called by the ledger (utility process).
### How it started:
1. add a callback type to the `ledger::client` namespace (`components/brave_rewards/core/ledger_client.h`):
    ```cpp
    using IdentityCallback = std::function<void(const std::string&)>;
    ```
2. add a pure `virtual` function to the `ledger::LedgerClient` interface (`components/brave_rewards/core/ledger_client.h`):
    ```cpp
    virtual void Identity(const std::string& value,
                          client::IdentityCallback callback) = 0;
    ```
3. `brave_rewards::RewardsServiceImpl` has to provide its implementation of `Identity()` (as it implements `ledger::LedgerClient`):
    - `components/brave_rewards/browser/rewards_service_impl.h`:
        ```cpp
        void Identity(const std::string& value,
                      ledger::client::IdentityCallback callback) override;
        ```
    - `components/brave_rewards/browser/rewards_service_impl.cc`:
        ```cpp
        void RewardsServiceImpl::Identity(const std::string& value,
                                          ledger::client::IdentityCallback callback) {
          callback(value);
        }
        ```
4. now add an interface method to `bat_ledger.mojom.BatLedgerClient` (`components/services/bat_ledger/public/interfaces/bat_ledger.mojom`):
    ```
    Identity(string value) => (string result);
    ```
5. the generated `virtual` function has to be overridden in `bat_ledger::LedgerClientMojoBridge`, as this is the class that implements `bat_ledger::mojom::BatLedgerClient` (a pointer to which gets passed to the `mojo::AssociatedReceiver<bat_ledger::mojom::BatLedgerClient>` in `brave_rewards::RewardsServiceImpl`). It converts `base::OnceCallback<>`s to `std::function<>`s, and functions as a proxy for `ledger::LedgerClient`:
    - `components/services/bat_ledger/public/cpp/ledger_client_mojo_bridge.h`:
        ```cpp
        void Identity(const std::string& value, IdentityCallback callback) override;

        static void OnIdentity(CallbackHolder<IdentityCallback>* holder,
                               const std::string& result);
        ```
    - `components/services/bat_ledger/public/cpp/ledger_client_mojo_bridge.cc`:
        ```cpp
        void LedgerClientMojoBridge::Identity(const std::string& value,
                                              IdentityCallback callback) {
          // note that this is the deprecated way of converting
          // from `base::OnceCallback<>` to `std::function<>`,
          // as the callback held by `holder` is leaked
          // if the bind expression below does not get called for some reason
          auto* holder =
              new CallbackHolder<IdentityCallback>(AsWeakPtr(), std::move(callback));
          ledger_client_->Identity(
              value, std::bind(LedgerClientMojoBridge::OnIdentity, holder, _1));
        }

        void LedgerClientMojoBridge::OnIdentity(
            CallbackHolder<IdentityCallback>* holder,
            const std::string& result) {
          DCHECK(holder);
          if (holder->is_valid()) {
            std::move(holder->get()).Run(result);
          }

          delete holder;
        }
        ```
6. now comes the ledger: `bat_ledger::BatLedgerClientMojoBridge` is the abstraction on the other end that takes care of converting between callback types (`std::function<>`s to `base::OnceCallback<>`s this time), and calls the functions on the `mojo::AssociatedRemote<bat_ledger::mojom::BatLedgerClient>`. As such, it implements the `ledger::LedgerClient` interface, therefore needs an implementation of `Identity()` as well:
    - `components/services/bat_ledger/bat_ledger_client_mojo_bridge.h`:
        ```cpp
        void Identity(const std::string& value,
                      ledger::client::IdentityCallback callback) override;

        void OnIdentity(ledger::client::IdentityCallback callback,
                        const std::string& result);
        ```
    - `components/services/bat_ledger/bat_ledger_client_mojo_bridge.cc`:
        ```cpp
        void BatLedgerClientMojoBridge::Identity(
            const std::string& value,
            ledger::client::IdentityCallback callback) {
          bat_ledger_client_->Identity(
              value, base::BindOnce(&BatLedgerClientMojoBridge::OnIdentity, AsWeakPtr(),
                                    std::move(callback)));
        }

        void BatLedgerClientMojoBridge::OnIdentity(
            ledger::client::IdentityCallback callback,
            const std::string& result) {
          callback(result);
        }
        ```
Same goes for `bat_ledger.mojom.BatLedger`.

### How it's going:
1. add an interface method to `ledger.mojom.LedgerClient` (`components/brave_rewards/common/mojom/bat_ledger.mojom`):
    ```
    Identity(string value) => (string result);
    ```
2. add its implementation to `brave_rewards::RewardsServiceImpl`:
    - `components/brave_rewards/browser/rewards_service_impl.h`:
        ```cpp
        void Identity(const std::string& value, IdentityCallback callback) override;
        ```
    - `components/brave_rewards/browser/rewards_service_impl.cc`:
        ```cpp
        void RewardsServiceImpl::Identity(const std::string& value,
                                          IdentityCallback callback) {
          std::move(callback).Run(value);
        }
        ```
Same goes for `ledger.mojom.Ledger`.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
QA's Rewards smoke test plan (@LaurenWags):
- [ ] Verify that none of the reward endpoints are being contacted when a user visits a media creator (`youtube.com`, `reddit.com`, `twitter.com`, `github.com`) and hasn't joined rewards
  - [ ] Verify that `rewards.brave.com`, `pcdn.brave.com`, `grant.rewards.brave.com` or `api.rewards.brave.com` are not being contacted
- [ ] Verify you are able to create a new Rewards profile and are in the unverified state by default
  - [ ] Verify when you visit a creator in this state the panel shows a prompt to connect a custodian and no BAT information (earnings, balance, etc) is displayed
  - [ ] Verify when you click on an in-line tip button in this state, the tip banner indicates you must connect a custodian
  - [ ] Verify when you view the NTP widget in this state it shows a prompt to connect a custodian and no BAT information (earnings, balance, etc) is displayed
  - [ ] Verify when you visit brave://rewards/ in this state there are no Auto Contribution, Tipping, or summary panels on this page
  - [ ] Verify you can toggle ads off/on from this page and the panel and NTP widget respond accordingly
  - [ ] Verify you can toggle rewards off/on from the panel and the brave://rewards/ page and NTP widget respond accordingly
- [ ] Verify you are able to connect a custodian
  - [ ] Verify Rewards balance shows correct BAT and USD value on brave://rewards/, panel, and NTP widget after you connect
  - [ ] Verify when you click on the BR panel while on a site, the panel displays site-specific information (site favicon, domain, attention %)
  - [ ] Verify BR panel and tip banner shows message about an unverified creator
  - [ ] Verify BR panel and tip banner shows message about a creator with a different custodian
  - [ ] Verify you are able to tip a creator who has the same custodian as you
  - [ ] Verify you are able to perform an auto contribution
  - [ ] Verify if you disable auto-contribute you are still able to tip creators
  - [ ] Verify if auto-contribute is disabled AC does not occur
  - [ ] Verify that there is no "Disconnect" option once connected to a custodian
- [ ] Verify that you are able to reset rewards
  - [ ] Verify after rewards are reset, the panel, brave://rewards/ page, and NTP widget to back to the unverified state

### CODE REVIEW

File: ios/browser/api/ledger/ledger_client_ios.mm [hunk 1]
```diff
@@ -175,17 +219,23 @@
...
-    const std::string& value) {
-  return [bridge_ decryptString:value];
+void LedgerClientIOS::EncryptString(const std::string& value,
+                                    EncryptStringCallback callback) {
+  [bridge_ encryptString:value callback:std::move(callback)];
 }
+void LedgerClientIOS::DecryptString(const std::string& value,
+                                    DecryptStringCallback callback) {
+  [bridge_ decryptString:value callback:std::move(callback)];
+}
```
Comment (szilardszaloki) [inline ios/browser/api/ledger/ledger_client_ios.mm:241]: newline

File: ios/browser/api/ledger/brave_ledger.mm [hunk 2]
```diff
@@ -1029,72 +914,61 @@ - (void)onReconcileComplete:(ledger::mojom::Result)result
...
+    ledger->GetAutoContributeProperties(
+        base::BindOnce(^(ledger::mojom::AutoContributePropertiesPtr props) {
+          auto properties =
+              props.get() != nullptr
+                  ? [[LedgerAutoContributeProperties alloc]
+                        initWithAutoContributePropertiesPtr:std::move(props)]
+                  : nil;
+          dispatch_async(dispatch_get_main_queue(), ^{
+            completion(properties);
+          });
```
Comment (szilardszaloki) [inline ios/browser/api/ledger/brave_ledger.mm:969]: Curious: couldn't we achieve the same with `PostTaskAndReplyWithResult()`?

Comment (szilardszaloki) [reply ios/browser/api/ledger/brave_ledger.mm:969]: Discussed in DM.

File: components/services/bat_ads/public/interfaces/BUILD.gn [hunk 3]
```diff
@@ -6,8 +6,6 @@
 import("//mojo/public/tools/bindings/mojom.gni")
 
 mojom("interfaces") {
```
Comment (zenparsing) [inline components/services/bat_ads/public/interfaces/BUILD.gn:8]: I'd probably leave these changes to a separate PR.

File: ios/browser/api/ads/brave_ads.mm [hunk 4]
```diff
@@ -310,7 +310,11 @@ - (void)shutdown:(nullable void (^)())completion {
           delete self->adsClient;
         }
         if (self->adsDatabase != nil) {
-          delete self->adsDatabase;
+          self->databaseQueue->PostTask(
+              FROM_HERE,
+              base::BindOnce(
+                  [](brave_ads::Database* database) { delete database; },
```
Comment (zenparsing) [inline ios/browser/api/ads/brave_ads.mm:316]: Should this be in a separate ads PR?

File: ios/browser/api/common/common_operations.mm [hunk 5]
```diff
@@ -105,46 +117,43 @@ - (void)loadURLRequest:(const std::string&)url
...
-          auto* responseHeaders =
-              new base::flat_map<std::string, std::string>();
-          [response.allHeaderFields
-              enumerateKeysAndObjectsUsingBlock:^(NSString* _Nonnull key,
-                                                  NSString* _Nonnull obj,
-                                                  BOOL* _Nonnull stop) {
-                if (![key isKindOfClass:NSString.class] ||
-                    ![obj isKindOfClass:NSString.class]) {
-                  return;
+          strongSelf->_taskRunner->PostTask(
```
Comment (zenparsing) [inline ios/browser/api/common/common_operations.mm:120]: Why is this change required?

File: ios/browser/api/ledger/brave_ledger.mm [hunk 6]
```diff
@@ -43,26 +43,41 @@
 #error "This file requires ARC support."
 #endif
 
-#define BLOG(verbose_level, format, ...)                  \
+namespace ledger {
+
+template <typename T>
+struct task_deleter {
```
Comment (zenparsing) [inline ios/browser/api/ledger/brave_ledger.mm:49]: Is there a built-in Chromium thing that does this?

File: ios/browser/api/ledger/brave_ledger.mm [hunk 7]
```diff
@@ -43,26 +43,41 @@
...
+};
+
+template <typename T, typename... Args>
+std::unique_ptr<T, task_deleter<T>> make_task_ptr(Args&&... args) {
+  return {new T(std::forward<Args>(args)...), task_deleter<T>()};
+}
+
+}  // namespace ledger
+
+#define LLOG(verbose_level, format, ...)                  \
```
Comment (zenparsing) [inline ios/browser/api/ledger/brave_ledger.mm:75]: Why the change to `LLOG`?

File: ios/browser/api/ledger/brave_ledger.mm [hunk 8]
```diff
@@ -108,10 +123,15 @@ typedef NS_ENUM(NSInteger, BATLedgerDatabaseMigrationType) {
 };
 
 @interface BraveLedger () <LedgerClientBridge> {
-  LedgerClientIOS* ledgerClient;
-  ledger::Ledger* ledger;
+  // DO NOT ACCESS DIRECTLY, use `postLedgerTask` or ensure you are accessing
+  // _ledger from a task posted in `_ledgerTaskRunner`
+  std::unique_ptr<ledger::LedgerImpl, ledger::task_deleter<ledger::LedgerImpl>>
```
Comment (zenparsing) [inline ios/browser/api/ledger/brave_ledger.mm:128]: Does this need to be a `LedgerImpl` pointer? Is it possible to avoid the `postLedgerTask` calls by creating a mojo remote that lives in the same process?

File: components/brave_rewards/common/mojom/bat_ledger.mojom [hunk 9]
```diff
@@ -1,52 +1,68 @@
...
 import "brave/components/brave_rewards/common/mojom/ledger.mojom";
 import "brave/components/brave_rewards/common/mojom/ledger_database.mojom";
 import "brave/components/brave_rewards/common/mojom/ledger_types.mojom";
 import "mojo/public/mojom/base/time.mojom";
 import "mojo/public/mojom/base/values.mojom";
 
-interface BatLedgerService {
-  Create(pending_associated_remote<BatLedgerClient> bat_ledger_client,
-         pending_associated_receiver<BatLedger> database) => ();
+interface Ledger {
```
Comment (zenparsing) [inline components/brave_rewards/common/mojom/bat_ledger.mojom:13]: I would love to see us move away from the "ledger" terminology. It seems like we are doubling down on it here, though. I understand that since we have "LedgerImpl" (which we can't change yet), it is intuitive to name this "Ledger". However, I think it would be better to go ahead and rename this to something better now (e.g. `RewardsEngine`, but we should throw some ideas around), rather that wait.

**Review by zenparsing — COMMENTED** [review]
Looking really great - thanks!

File: ios/browser/api/ledger/brave_ledger.mm (hunk 7 above)
Comment (kylehickinson) [reply ios/browser/api/ledger/brave_ledger.mm:75]: Because `BLOG` was already defined and this was a redefinition that used our own client stuff… I agree though it'd be better to the replace the LedgerClient log method with Chromium logging directly since they all use Chromium logging anyways

File: ios/browser/api/ledger/brave_ledger.mm (hunk 8 above)
Comment (kylehickinson) [reply ios/browser/api/ledger/brave_ledger.mm:128]: Nope not possible because of `[Sync]` usages in mojom + ledger at the moment. `[Sync]` type calls are not allowed on the main thread on iOS and iOS is single-process only, so we can't spawn the whole implementation in a separate process to ensure it never uses main

File: ios/browser/api/ledger/brave_ledger.mm (hunk 6 above)
Comment (kylehickinson) [reply ios/browser/api/ledger/brave_ledger.mm:49]: Not as far as I'm aware of

File: ios/browser/api/common/common_operations.mm (hunk 5 above)
Comment (kylehickinson) [reply ios/browser/api/common/common_operations.mm:120]: Because the callbacks into ledger have to be called on the same task runner that ledger runs on

File: components/brave_rewards/browser/BUILD.gn [hunk 10]
```diff
@@ -46,7 +46,7 @@ static_library("browser") {
     "//brave/components/brave_rewards/common",
     "//brave/components/brave_rewards/common/buildflags",
     "//brave/components/resources",
-    "//brave/components/services/bat_ledger/public/cpp",
+    "//brave/components/services/ledger/public/interfaces",
```
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/browser/BUILD.gn:49]: Sure. `brave` is already in the path, so I don't think it would be necessary to add `brave_` as a prefix. Ultimately, I'd like to end up with `brave/components/services/rewards_engine`, and `RewardsEngineService` in `rewards_engine_service.mojom`. Now that it'll host `LedgerImpl`s (`RewardsEngineImpl`s) for multiple profiles (coming up soon), I realize `LedgerFactory` is not the best name. That said, I'd rather not spend more time renaming folders/classes/etc. in this PR. Is it okay with you to do it in a follow-up?

File: components/brave_rewards/common/mojom/bat_ledger.mojom [hunk 11]
```diff
@@ -1,52 +1,68 @@
 // Copyright (c) 2019 The Brave Authors. All rights reserved.
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
-// You can obtain one at http://mozilla.org/MPL/2.0/.
-module bat_ledger.mojom;
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+module ledger.mojom;
```
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/common/mojom/bat_ledger.mojom:5]: I'd do this in the follow-up mentioned above. Since `brave_rewards` is already in the path, would `rewards.mojom` work for you?

File: components/services/bat_ads/public/interfaces/BUILD.gn (hunk 3 above)
Comment (szilardszaloki) [reply components/services/bat_ads/public/interfaces/BUILD.gn:8]: Doing this as per @tmancey's request — I can withdraw this change and file an issue for @brave/ads-client.

File: components/brave_rewards/common/mojom/bat_ledger.mojom (hunk 11 above)
[the remark this answers is not in the document]
Comment (zenparsing) [reply components/brave_rewards/common/mojom/bat_ledger.mojom:5]: Sorry, I was thinking about the generated C++ namespace instead of the file name. If we are going to move everything into the `brave_rewards` namespace, could we just put this into the `brave_rewards::mojom` namespace now?

File: components/brave_rewards/browser/BUILD.gn (hunk 10 above)
[the remark this answers is not in the document]
Comment (zenparsing) [reply components/brave_rewards/browser/BUILD.gn:49]: If we're going to rename the folder later, then we should just leave this folder as `bat_ledger` here, I think.

File: components/brave_rewards/common/mojom/bat_ledger.mojom (hunk 11 above)
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/common/mojom/bat_ledger.mojom:5]: The problem with doing that in this PR is that we have this pattern:
```
namespace ledger {
  ... `mojom::something`
}
```
Above works because of `module ledger.mojom;` (since `mojom` is a nested namespace of `ledger`). If I changed this now to `module brave_rewards.mojom;`, I'd have to replace the above usages with
```
namespace ledger {
  ... `brave_rewards::mojom::something`
}
```
unless I changed the `ledger` namespace everywhere to `brave_rewards`, which would explode this PR even more.

[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/common/mojom/bat_ledger.mojom:5]: Also, ideally, namespace names should be short. `brave_rewards` is very long. Any particular reason why we're not considering `rewards`?

File: components/brave_rewards/core/wallet/wallet_utils_unittest.cc [hunk 12]
```diff
@@ -22,12 +26,10 @@ namespace ledger::wallet {
 mojom::ExternalWalletPtr ExternalWalletPtrFromJSON(std::string wallet_string,
                                                    std::string wallet_type);
 
-class WalletUtilTest : public BATLedgerTest {};
+class WalletUtilTest : public Test {};
```
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/core/wallet/wallet_utils_unittest.cc:29]: `WalletUtilTest` doesn't need a ledger environment.

File: components/brave_rewards/browser/BUILD.gn (hunk 10 above)
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/browser/BUILD.gn]: Done (https://github.com/brave/brave-core/pull/17734/commits/e8bcdc7e5858af65c9a0c72353e3ce54e788aa15)

File: components/brave_rewards/core/endpoint/private_cdn/get_publisher/get_publisher_unittest.cc [hunk 13]
```diff
@@ -7,41 +7,22 @@
...
-                   *info = std::move(request_info);
-                   run_loop.Quit();
-                 });
-
-    run_loop.Run();
-    return result;
-  }
+namespace ledger::endpoint::private_cdn {
 
+class GetPublisherTest : public testing::Test {
```
[the remark this answers is not in the document]
Comment (szilardszaloki) [reply components/brave_rewards/core/endpoint/private_cdn/get_publisher/get_publisher_unittest.cc]: Done (https://github.com/brave/brave-core/pull/17734/commits/e8bcdc7e5858af65c9a0c72353e3ce54e788aa15)

File: components/services/bat_ads/public/interfaces/BUILD.gn (hunk 3 above)
Comment (szilardszaloki) [reply components/services/bat_ads/public/interfaces/BUILD.gn]: Done (https://github.com/brave/brave-core/pull/17734/commits/e8bcdc7e5858af65c9a0c72353e3ce54e788aa15)

File: components/brave_rewards/common/mojom/bat_ledger.mojom (hunk 11 above)
[the remark this answers is not in the document]
Comment (zenparsing) [reply components/brave_rewards/common/mojom/bat_ledger.mojom:5]: Your point above makes sense 👍

**Review by zenparsing — APPROVED** [review]

File: components/services/bat_ads/public/interfaces/BUILD.gn (hunk 3 above)
Comment (szilardszaloki) [reply components/services/bat_ads/public/interfaces/BUILD.gn]: Filed an issue [here](https://github.com/brave/brave-browser/issues/29620).
cc: @brave/ads-client

File: ios/browser/api/ads/brave_ads.mm (hunk 4 above)
Comment (kylehickinson) [reply ios/browser/api/ads/brave_ads.mm:316]: @zenparsing Not 100% sure if this occurred outside of this PR, but definitely happened while I was testing my changes in this branch so I just tossed it in here as well. We can take it out and have it in a separate PR entirely if we want

Comment (zenparsing) [reply ios/browser/api/ads/brave_ads.mm:316]: No strong opinion here, just checking!

**Review by Brandon-T — APPROVED** [review]
